text/0003-meta-spec-v2.md

```diff
@@ -0,0 +1,1281 @@
+    "platforms": [
+      "linux-amd64", "linux-arm64",
+      "darwin-amd64", "darwin-arm64"
+    ],
+    "packages": {
+      "configure": {
+        "requires": {
+          "pkg:cargo/cargo-pgrx": "==0.11.4",
+          "pkg:generic/bison": 0,
+          "pkg:generic/cmake": 0,
+          "pkg:generic/flex": 0,
+          "pkg:generic/readline": 0,
+          "pkg:generic/openssl": 0,
+          "pkg:generic/pkg-config": 0
+        }
+      },
+      "run": {
+        "requires": {
+            "pkg:generic/penblas": 0,
+            "pkg:generic/python3": 0,
+            "pkg:generic/readline": 0,
+            "pkg:generic/openssl": 0,
+            "pkg:generic/bison": 0
+        },
+        "recommends": {
+            "pkg:pypi/pyarrow": "11.0.0",
+            "pkg:pypi/catboost": 0,
+            "pkg:pypi/lightgbm": 0,
+            "pkg:pypi/torch": 0,
+            "pkg:pypi/langchain": 0
+        }
+      }
+    },
+    "variations": [
+      {
+        "where": {
+          "platforms": ["linux"]
+        },
+        "dependencies": {
+          "packages": {
+            "run": {
+              "recommends": {
+                "pkg:pypi/auto-gptq": 0,
+                "pkg:pypi/xformers": 0
+              }
+            }
+          }
+        }
+      }
+    ]
+  }
+#}
+```
+
+(Spec 2) [optional] {[Object](#object)}
+
+This property identifies dependencies required to configure, build, test,
+install, and run the [Package](#package) provided by the
+[Distribution](#source-distribution). These include not only PGXN packages,
+but also external libraries, system dependencies, and versions of PostgreSQL
+--- as well as any OS and architectures ([arm64], [amd64], etc.).
+
+[Consumers](#consumer) **SHOULD** use this data to determine what dependencies
+to install.
+
+Properties:
+
+*   **platforms**: An [Array](#array) of one or more [Platform](#platform)
+    strings that identify OSes and architectures supported by the
+    [Package](#package) provided by the [Distribution](#source-distribution).
+    If this property is not present, [Consumers](#consumer) **SHOULD** assume
+    that the [Package](#package) supports any platform that PostgreSQL
+    supports. This property is typically needed only when the
+    [Package](#package) depends on platform-specific features.
+
+*   **postgres**: An [Object](#object) describing the versions of PostgreSQL
+    required by the [Package](#package) provided by the
+    [Distribution](#source-distribution). The object supports the following
+    properties:
+    *   **version**: A [Version Range](#version-range) identifying the
+        supported versions of PostgreSQL. **REQUIRED**.
+    *   **with**: An [Array](#array) of [Terms](#term) that specify features
+        that are required to be compiled into PostgreSQL. Each corresponds to
+        the appropriate `--with` [configure flag]. **OPTIONAL**.
+
+*   **pipeline**: A [Term](#term) identifying the build pipeline required to
```

Comment:
I'm wondering if this concept needs to be extended to support additional paraemters. Some examples from recent experience rebuilding Trunk extensions:

```
*   `cargo pgrx package --features icu`
*   `RUSTFLAGS="-C target-feature=+avx2,+fma" cargo pgrx package`
*   `mvn clean install -Dpljava.libjvmdefault=/usr/lib/x86_64-linux-gnu/libjvm.so`
```